privaterelay/tests/fxa_utils_tests.py

```diff
@@ -0,0 +1,188 @@
+"""
+Tests for private_relay/fxa_utils.py
+"""
+from datetime import datetime, timedelta, timezone
+from unittest.mock import patch
+import pytest
+
+from django.conf import settings
+
+from allauth.socialaccount.models import SocialAccount
+from model_bakery import baker
+from privaterelay.fxa_utils import get_phone_subscription_dates
+from waffle.models import Flag
+
+pytestmark = pytest.mark.skipif(
+    not settings.PHONES_ENABLED, reason="PHONES_ENABLED is False"
+)
+
+if settings.PHONES_ENABLED:
+    from phones.tests.models_tests import make_phone_test_user
+
+
+MOCK_BASE = "privaterelay.fxa_utils"
+
+
+@pytest.fixture()
+def phone_user(db):
+    yield make_phone_test_user()
+
+
+@patch(f"{MOCK_BASE}.get_subscription_data_from_fxa")
+def test_get_phone_subscription_dates_refreshed_twice(mocked_phone_dates, phone_user):
+    social_account = SocialAccount.objects.get(user=phone_user)
+    mocked_phone_dates.return_value = {"social_token": "new token", "refreshed": True}
```

Comment:
suggestion (blocking): it seems like this `mocked_phone_dates` variable name should be `mocked_data_from_fxa`?

Or maybe this is an indication that the `get_subscription_data_from_fxa` is returning 2 very different things in different situations? I.e., if the token is expired or the requested scopes are not allowed, it returns a dictionary with a new token and a "refreshed" indicator. But if the token is valid, then it returns the data from FXA.

That smells like this function is doing 2 very different things and returning 2 very different kinds of data.

Reply:
No you are correct and I updated the tests to use the new variable name `mocked_data_from_fxa` on 73858e9. Agreed on potentially confusing `get_subscription_data_from_fxa` but with the refresh also using the client it's a bit complicated to refactor now and hopefully when we revisit the refresh logic we can handle it there.